src/PactNet.Native/Messaging/MessageMiddleware.cs

```diff
@@ -0,0 +1,91 @@
+#if NETSTANDARD2_0_OR_GREATER
+
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+using Microsoft.AspNetCore.Http;
+
+using Newtonsoft.Json;
+
+namespace PactNet.Native.Messaging
+{
+    /// <summary>
+    /// Defines the message middleware
+    /// It handles the queries for the verifier when there are message interactions
+    /// </summary>
+    public class MessageMiddleware
+    {
+
+        private readonly RequestDelegate _next;
+
+        /// <summary>
+        /// Creates an instance of <see cref="MessageMiddleware"/>
+        /// </summary>
+        /// <param name="next"></param>
+        public MessageMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        /// <summary>
+        /// The invoke method of the middleware
+        /// </summary>
+        /// <param name="context">The http context</param>
+        /// <returns>A task</returns>
+        public async Task InvokeAsync(HttpContext context)
+        {
+            if (!(context.Request.Path.Value?.StartsWith(Constants.PactMessagesPath) ?? false))
+            {
+                await _next(context);
+                return;
+            }
+
+            context.Response.StatusCode = (int)HttpStatusCode.OK;
+
+            string jsonRequestBody = await GetRequestBodyAsync(context);
+
+            var interactionDescription = JsonConvert.DeserializeObject<MessageInteraction>(jsonRequestBody)?.Description;
+
+            if (string.IsNullOrWhiteSpace(interactionDescription))
+            {
+                //See if raising an exception is better
```

Comment:
Yeah I think it's definitely better to throw a 500 error instead of just silently failing.

Reply:
I'm not sure how to handle exceptions on middleware. I'll research and will add it on a later PR